### DISCUSSION TITLE
GC9A01 fails to initialize consistently, any advice?

### DISCUSSION THREAD

**Discussion by rsiemens77** [Bodmer/TFT_eSPI#3173] [Compatible displays and setup files]
I am using the latest 2.5.33 library, along with ESP32 processor, IDFv4.4.6 environment, and GC9A01 1.28" round SPI LCD (240x240), like the round PCB one here:

https://www.aliexpress.us/item/3256805909598124.html

I have the LCD reset connected to the ESP32 reset line, set to -1 in the config.  I have also tried reset linked to an IO, didn't make a difference.

I previously started on an ST7789 rectangular LCD (240x320) and everything worked perfectly.  I now switched to the above GC9A01, and it often doesn't work: it appears to fail at initialization.  I repeatedly reset, and after a random number of times, it will work.  There does not seem to be a flaky connection, wiggling wires does not seem to make a difference.  I tried different frequencies, slower ones, does not seem to effect the behavior.

Often on failure it looks like garbage updating on the LCD.  But if it gets past initialization, it works perfectly.

Any thoughts on what might be causing this?  Is there anything timing/synchronization critical in initialization?  Anything else I can test?

Or is there likely something wrong with my LCD?  I only have one GC9A01 to test with.  As I mentioned, this all works perfectly on the ST7789 I have.  Is my best bet just to order another GC9A01 to see if that's the issue?

**Comment by Bodmer** [comment]
Try reducing the SPI clock speed in the setup file.
